File: scripts/sum.c
#include <stdio.h>
#include <stdlib.h>


int is_num_integer(const char *s)
{
    while (*s) {
        if (*s<'0' || *s>'9') return 0;
        *s++;
    }
    // return 1 if all chars are digits
    return 1;
}

int main(int argc, char * argv[]) 
{     
    int sum;	

    if (argc != 3) 
    {
        printf("Enter only two arguments..\n");
        return 1;
    }  

    if( is_num_integer(argv[1]) == 0 || is_num_integer(argv[2]) == 0)
    {
        printf("Please enter only integer numbers..\n");
        return 2;
    }    

    // atoi: convert string to integer
    sum = atoi(argv[1]) + atoi(argv[2]); 
    printf("%d\n", sum);  

    return 0;
}

File: scripts/sum.py
import sys

# This program print the sum of two integer numbers 
# Get the numbers from user arguments like:  ./sum.py 33 44 => print 77

# Check that user sends only 2 numbers as arguments, I write 3 in if statement
# because the first argument is for the program file name + 2 arguments (numbers)
if len(sys.argv) != 3:
    print("Enter only two arguments..")
    sys.exit(1)

num1 = sys.argv[1]
num2 = sys.argv[2]

    
# Check if the two arguments are numbers    
if not num1.isdigit() or not num2.isdigit():
    print("Please enter only integer numbers..")
    sys.exit(2)
 
print(f"{int(num1) + int(num2)}")
